lib/net/nvnetdrv/nvnetdrv_lwip.c

```diff
@@ -123,6 +134,13 @@ static err_t low_level_init (struct netif *netif)
     /* set MAC hardware address */
     memcpy(netif->hwaddr, nvnetdrv_get_ethernet_addr(), 6);
 
+    // Let LWIP know the NIC is initialized and ready to use
+    netif_set_up(g_pnetif);
+
+    // Let LWIP know if a ethernet cable is currently connected
+    bool link_up = nvnetdrv_is_link_up();
+    change_link_state((void *)link_up);
```

Comment:
Should (or can?) this happen before `netif_set_up`?

If the link state defaults to `up` and no cable is connected, lwip would first set up the NIC with default link state `up`, and then immediately change the link state `down` again.